libs/libskynet/src/encrypt.ts

```diff
@@ -0,0 +1,58 @@
+import { encodeU64 } from "./encoding.js";
+import { SHA512_HASH_SIZE, sha512 } from "./sha512.js";
+
+// otpEncrypt takes a key and some data and encrypts the data with the key. The
+// encryption happens by generating a sequence of bytes using sha512 hashes and
+// then xor'ing those bytes with the data. This gives otpEncrypt similar
+// security properties to a one-time-pad - which means that the same key cannot
+// be used twice!
+//
+// It also means that there is no authentication on the data, and that an
+// attacker could flip bits undetected if an authentication layer is not added
+// on top.
+//
+// Data is encrypted in-place. The optional value 'skip' allows the caller to
+// specify a number of bytes to skip initially.
+//
+// NOTE: otpEncrypt can be useful over other encryption methods because it does
+// not introduce any new dependencies. For the Skynet Kernel bootloader, the
+// only cryptography present is ed25519 signatures (which includes sha512 as a
+// dependency). This is a tiny piece of code that can provide encryption
+// support without needing to add a full encryption library as a dependency.
+//
+// WARNING: otpEncrypt is not a "safe" function. It's a useful cryptographic
+// primitive, but there are easy ways to misuse it and create insecure
+// encryption schemes. Please avoid using this function unless you have a
+// strong understanding of encryption techniques and typical encryption
+// attacks.
+function otpEncrypt(key: Uint8Array, data: Uint8Array, skip?: number): Uint8Array {
+  // If the optional variable is not set, set it.
+  if (skip === undefined) {
+    skip = 0;
+  }
```

Comment:
`skip` parameter could default to 0:

```suggestion
function otpEncrypt(key: Uint8Array, data: Uint8Array, skip: number = 0): Uint8Array {
```